editor.planx.uk/src/pages/FlowEditor/components/FlowEditor/ToggleDataFieldsButton.tsx

```diff
@@ -0,0 +1,44 @@
+import DataFieldIcon from "@mui/icons-material/AutoFixNormal";
+import DataFieldOffIcon from "@mui/icons-material/AutoFixOff";
+import Box from "@mui/material/Box";
+import IconButton from "@mui/material/IconButton";
+import Tooltip from "@mui/material/Tooltip";
+import { useStore } from "pages/FlowEditor/lib/store";
+import React from "react";
+
+export const ToggleDataFieldsButton: React.FC = () => {
+  const [showDataFields, toggleShowDataFields] = useStore((state) => [
+    state.showDataFields,
+    state.toggleShowDataFields,
+  ]);
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "fixed",
+        bottom: theme.spacing(6),
+        left: theme.spacing(7),
+        zIndex: theme.zIndex.appBar,
+        border: `1px solid ${theme.palette.border.main}`,
+        borderRadius: "3px",
+        background: theme.palette.background.paper,
+      })}
+    >
+      <Tooltip title="Toggle data fields" placement="right">
+        <IconButton
+          aria-label="Toggle data fields"
+          onClick={toggleShowDataFields}
+          size="large"
+          sx={(theme) => ({
+            padding: theme.spacing(1),
+            color: showDataFields
+              ? theme.palette.text.primary
+              : theme.palette.text.disabled,
+          })}
+        >
+          {showDataFields ? <DataFieldIcon /> : <DataFieldOffIcon />}
```

Comment:
Open to MUI icon suggestions for this one especially - there's only so many that have "on"/"off" pairings ! 

But I think there's something sweet and hopefully not too far of a stretch about "auto" wand &rarr; "auto-answering" &rarr; data fields ?

Reply:
I'd say we need to be more literal as the magic element is an implied function of the data itself, rather than describing the function of the button.

Not perfect from a technical sense (code vs data), but `CodeIcon` and `CodeOffIcon` is a really simple icon combo that gives a stronger literal link to the 'thing' being toggled.

Reply:
Good shout - that's definitely a clearer option - updated!